src/state/selectors/combiners/combineTooltipInteractionState.ts

```diff
@@ -0,0 +1,62 @@
+import { noInteraction, TooltipIndex, TooltipInteractionState, TooltipState } from '../../tooltipSlice';
+import { TooltipEventType } from '../../../util/types';
+import { TooltipTrigger } from '../../../chart/types';
+
+function chooseAppropriateMouseInteraction(
+  tooltipState: TooltipState,
+  tooltipEventType: TooltipEventType,
+  trigger: TooltipTrigger,
+): TooltipInteractionState | undefined {
+  if (tooltipEventType === 'axis') {
+    if (trigger === 'click') {
+      return tooltipState.axisInteraction.click;
+    }
+    return tooltipState.axisInteraction.hover;
+  }
+  if (trigger === 'click') {
+    return tooltipState.itemInteraction.click;
+  }
+  return tooltipState.itemInteraction.hover;
+}
+
+function hasBeenActivePreviously(tooltipInteractionState: TooltipInteractionState): boolean {
+  return tooltipInteractionState.index != null;
+}
+
+export const combineTooltipInteractionState = (
+  tooltipState: TooltipState,
+  tooltipEventType: TooltipEventType | undefined,
+  trigger: TooltipTrigger,
+  defaultIndex: TooltipIndex | undefined,
+): TooltipInteractionState => {
+  if (tooltipState.syncInteraction.active && tooltipState.syncInteraction.index != null) {
```

Comment:
Now we can go here and update the synchronisation preference logic and all the other selectors will follow the same.